aopy/preproc/quality.py

```diff
@@ -35,12 +37,77 @@ def bad_channel_detection(data, srate, lf_c=100., sg_win_t=8., sg_over_t=4., sg_
 
     psd_var = np.var(Sxx_low_psd,axis=0)
     norm_psd_var = psd_var/npla.norm(psd_var)
-    low_var_θ = np.mean(norm_psd_var)/3
+    low_var_θ = np.mean(norm_psd_var)/num_th
     bad_ch_mask = norm_psd_var <= low_var_θ
 
     return bad_ch_mask
 
 
+def screenBadECoGchannels(data, th=0.05, numsd=5.0, debug=False, verbose=True):
+    '''
+    Detect badchannels. This code is originally Amy's code in pesaran lab.
+    This function has 2 steps. In the 1st step, it detects tentative bad channels whose SD is less than th or more than 1-th in CDF
+    In the 2nd step, it extracts bad channels from tentative bad channels by finding conditions where (SD - median of SD) is outside numsd*SD[~badch]
+    
+    Args:
+        data (nt, nch): neural data
+        th (float, optional): threshold to detect bad channels for 1st screening
+        numsd (float, optional): number of standard deviations above zero to detect bad channels for 2nd screening
+        debug (bool, optional): if True, display a figure showing the threshold crossings
+        verbose (bool, optional): if True, print bad channels
+        
+    Returns:
+        bad_ch (nch) : logical array indicating bad channels
+
+    Examples:
+        
+        .. code-block:: python
+        
+        test_data = np.random.normal(10,0.5,(10000, 200))
+        test_data[0, 10] = 25
+        test_data[5, 150] = 30
+        bad_ch = screenBadECoGchannels(test_data, th=0.05, numsd=5.0, debug=True, verbose=False)
+        
+        .. image:: _images/screenBadECoGchannels.png
+    '''
+    
+    assert th > 0 and th < 1, "Threshold must be between 0 and 1"
+    assert numsd > 0, "numsd must be more than 0"
+    
+    sd = np.std(data, axis=0)
+    med_sd = np.median(sd)
+    
+    nbins = 10000
+    hist, bins = np.histogram(sd, nbins)
```

Comment:
why this number of bins? can you make it an optional argument? or maybe it can be a multiple of the number of channels?